src/us/kbase/workspace/database/provenance/Common.java

```diff
@@ -47,38 +52,53 @@ static <T> List<T> processSimpleList(final List<T> list, final String name) {
 			return Common.immutable(list);
 		}
 	}
-	
+
+	static String checkRpid(final String putativeRpid, final String name) {
+		final String rpid = requireNonNull(processString(putativeRpid), name);
+
+		final Pattern validRpidRegex = Pattern.compile("^\\w+:\\w+");
+		final Matcher m = validRpidRegex.matcher(rpid);
+		if (m.find()) {
+			return rpid;
+		}
+		throw new IllegalArgumentException(String.format(
+				"Illegal ID format for %s: \"%s\"\n" + "ID prefixes should match the pattern \"^\\w+:\\w+\"", name,
+				putativeRpid));
```

Comment:
I would prefer the string on line 1 and the variables on line 2... oh but formatter, nm

Reply:
:+1: